.github/workflows/nuget-release.yml

```
@@ -0,0 +1,50 @@
+name: Create NuGet and GitHub Release
+
+on:
+  workflow_call:
+    inputs:
+      dotnet-version:
+        default: '6.0.x'
+        type: string
+      project-path:
+        required: true
+        type: string
+      tag-prefix:
+        default: ''
+        type: string
+      nuspec-file:
+        type: string
+        required: true
+    secrets:
+      github-token:
+        required: true
+      nuget-token:
+        required: true
+
+jobs:
+  release:
+    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'pull_request' && github.event.pull_request.merged && startsWith(github.event.pull_request.head.ref, 'release/'))
+    runs-on: windows-latest
+    environment: release
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - uses: actions/download-artifact@v4
```

Is it worth downgrading this to v3 as there seems to be issues amongst the latest *-artifact action releases
```suggestion
    - uses: actions/download-artifact@v3
```